lisp/main.lisp

```diff
@@ -33,33 +33,57 @@
     (new-view handle-new-view-event)
     (view-destroyed handle-view-destroyed-event)))
 
+(defun load-config-file (&optional (catch-errors t))
+  "Load the user's config file. Returns a values list: whether the file loaded (t if no
+rc files exist), the error if it didn't, and the config file that was
+loaded. When CATCH-ERRORS is nil, errors are left to be handled
+further up. "
+  (let* ((xdg-config
+           (probe-file (merge-pathnames #p"mahogany/init.lisp" (uiop:xdg-config-home))))
+	 (fallback-config
+	   (probe-file (merge-pathnames #p".config/mahogany/init.lisp" (user-homedir-pathname))))
+	 (config-file (or xdg-config fallback-config)))
+    (if config-file
+        (if catch-errors
+            (handler-case (load config-file)
+              (error (c) (values nil (format nil "~a" c) config-file))
```

Comment:
In addition to returning the error, I would log the error at the `:warn` level just so it isn't silently ignored.